tests/solana_tests/primitives.rs

```diff
@@ -599,6 +604,215 @@ fn truncate_uint(n: &mut U256, width: usize) {
+            let limit = BigUint::from(2_u32).pow(width).sub(1_u32);
+
+            // Generate a random number within the the range 0..2^N
+            let first_operand_rand = rng.gen_biguint(width.into());
+
+            // Calculate a number that when multiplied by first_operand_rand, the result will overflow N bits
+            let second_operand_rand = limit.div(&first_operand_rand);
+
+            let return_value = contract.function(
+                "mul",
+                &[
+                    ethabi::Token::Uint(U256::from_big_endian(&first_operand_rand.to_bytes_be())),
+                    ethabi::Token::Uint(U256::from_big_endian(&second_operand_rand.to_bytes_be())),
+                ],
+                &[],
+                None,
+            );
+            let res = first_operand_rand * second_operand_rand;
+
+            assert_eq!(
+                return_value,
+                vec![ethabi::Token::Uint(U256::from_big_endian(
+                    &res.to_bytes_be()
+                ))]
+            );
+        }
+    }
+}
+
+#[test]
+
+fn test_overflow_detect_signed() {
+    let mut rng = rand::thread_rng();
+    for width in (72..=256).step_by(8) {
+        let src = r#"
+        contract test {
+         
+            function mul(intN a, intN b) public returns (intN) {
+                
+                return a * b;
+                
+            }
+
+           
+
+        }"#
+        .replace("intN", &format!("int{}", width));
+        let mut contract = build_solidity(&src);
+
+        contract.constructor("test", &[]);
+
+        // Generate the max value to fit in N bits
+        let limit = BigInt::from(2_u32).pow(width - 1).add(1_u32);
+
+        // Generate a random number within the the range 0..2^N
+        let first_operand_rand = rng.gen_bigint((width - 1).into()).sub(1_u32);
+
+        // Calculate a number that when multiplied by first_operand_rand, the result will overflow N bits
+        let mut second_operand_rand = limit / &first_operand_rand;
+
+        if let Sign::Minus = second_operand_rand.sign() {
+            // Decrease by 1 if negative
+            second_operand_rand = second_operand_rand.sub(1);
+        } else {
+            // Increase by 1 if psotive
+            second_operand_rand = second_operand_rand.add(1);
+        }
+
+        let res = contract.function_must_fail(
+            "mul",
+            &[
+                ethabi::Token::Int(bigint_to_eth(
+                    &first_operand_rand,
+                    width.try_into().unwrap(),
+                )),
+                ethabi::Token::Int(bigint_to_eth(
+                    &second_operand_rand,
+                    width.try_into().unwrap(),
+                )),
+            ],
+            &[],
+            None,
+        );
+
+        assert_ne!(res, Ok(0));
+    }
+}
+
+#[test]
+
+fn test_overflow_detect_unsigned() {
+    let mut rng = rand::thread_rng();
+    for width in (72..=256).step_by(8) {
```

Comment:
what about 8..64?

Reply:
Because that will waste time testing intrinsic LLVM functions, the work starts from 72 bits.
There are 4 tests like this so that will be a lot of unnecessary testing